docs/technical_documents/open_space_decider.md

```diff
@@ -0,0 +1,175 @@
+# OPEN SPACE DECIDER
+
+# Introduction
+
+Apollo planning is a scenario based planning method, each driving use case is treated as a different driving scenario.
+
+open space decider is used to process related infomation and provide information for subsequent optimizers 
+
+# Where is the code
+
+Please refer [open space decider](https://github.com/ApolloAuto/apollo/modules/planning/tasks/deciders/open_space_decider/open_space_roi_decider.cc)
+
+# Code Reading
+
+## Open space roi decider
+
+1. Input : obstacles info \ vehicle info \ road info \ parking space info
+
+  - IN PARKING STAGE (roi_type == OpenSpaceRoiDeciderConfig::PARKING)
+
+    1. Check parking space id and parking space boundary,then get parking spot;
+    ```cpp
+        bool GetParkingSpot(Frame *const frame,
+                            std::array<common::math::Vec2d, 4> *vertices,
+                            hdmap::Path *nearby_path);
+    ```
+    2. Search target parking spot on path and check whether the distance between ADC and target parking spot is appropriate;
+    ```cpp
+        void SearchTargetParkingSpotOnPath(
+            const hdmap::Path &nearby_path,
+            hdmap::ParkingSpaceInfoConstPtr *target_parking_spot);
+    ``` 
+    3. Set ADC origin state,include ADC position and heading;
+    ```cpp
+        void SetOrigin(Frame *const frame,
+                        const std::array<common::math::Vec2d, 4> &vertices);
+    ```
+
+    4. Get parking spot(left top \ left down \ right down \ right top) points info, convert parking spot points coordinates to vehicle coorinates, according to the relative position and parking direction(inward or outward) to set parking end pose;
+
+    ```cpp  
+        void SetParkingSpotEndPose(
+            Frame *const frame, const std::array<common::math::Vec2d, 4> &vertices);
+    ```  
+    5. Get parking boundary: convert parking spot points coordinates to vehicle coorinates and get points' projections on reference line;
+    
+    ```cpp
+        bool GetParkingBoundary(Frame *const frame,
+                                const std::array<common::math::Vec2d, 4> &vertices,
+                                const hdmap::Path &nearby_path,
+                                std::vector<std::vector<common::math::Vec2d>>
+                                *const roi_parking_boundary);
+    ```
+    6. Get road boundary: it starts from parking space center line(center_line_s) and extends a distance(roi longitudinal range) to both side along the s direction to get start_s and end_s; search key points(the point on the left/right lane boundary is close to a curb corner) and anchor points(a start/end point or the point on path with large curvatures) in this roi range; those key points, anchor points is called boundary points and the line segements between those points is called roi parking boundary; 
```

Comment:
both side->maybe "both sides" or "either side"?
anchor points is -> "are"
those points is -> "are"